crates/bifrost/src/providers/replicated_loglet/remote_sequencer.rs

```diff
@@ -205,6 +207,76 @@ where
 
         Ok(connection)
     }
+
+    /// Attempts to find tail.
+    ///
+    /// This first tries to find tail by synchronizing with sequencer. If this failed
+    /// duo to sequencer not reachable, it will immediately try to find tail by querying
+    /// fmajority of loglet servers
+    pub async fn find_tail(&self) -> Result<TailState<LogletOffset>, OperationError> {
+        // try to sync with sequencer
+        if self.sync_sequencer_tail().await.is_ok() {
+            return Ok(*self.known_global_tail.get());
+        }
+
+        // otherwise we need to try to fetch this from the log servers.
+        self.sync_log_servers_tail().await?;
+        Ok(*self.known_global_tail.get())
```

Comment:
Is there value in racing these two variants?

Reply:
I am not sure I get your question here.

Reply:
Whether we should run both variants for obtaining the known global tail in parallel/concurrently?

Reply:
Thank you for clarification. Yeah, that's definitely a good idea.